linodecli/baked/parsing.py

```diff
@@ -0,0 +1,184 @@
+            functools.partial(
+                lambda style, match: f"[{style}]{match['text']}[/]", style
+            ),
+            result,
+        )
+
+    return result
+
+
+def extract_markdown_links(description: str) -> Tuple[str, List[str]]:
+    """
+    Extracts all Markdown links from the given description and
+    returns them alongside the stripped description.
+
+    :param description: The description of a CLI argument.
+    :type description: str
+
+    :returns: The stripped description and a list of extracted links.
+    :rtype: Tuple[str, List[str]]
+    """
+    result_links = []
+
+    def _sub_handler(match: re.Match) -> str:
+        link = match["link"]
+        if link.startswith("/"):
+            link = f"https://linode.com{link}"
+
+        result_links.append(link)
+        return match["text"]
+
+    result_description = REGEX_MARKDOWN_LINK.sub(_sub_handler, description)
+
+    return result_description, result_links
+
+
+def get_short_description(description: str) -> str:
+    """
+    Gets the first relevant sentence in the given description.
+
+    :param description: The description of a CLI argument.
+    :type description: str
+
+    :returns: A single sentence from the description.
+    :rtype: set
+    """
+
+    target_lines = description.splitlines()
+    relevant_lines = None
+
+    for i, line in enumerate(target_lines):
+        # Edge case for descriptions starting with a note
+        if line.lower().startswith("__note__"):
+            continue
+
+        relevant_lines = target_lines[i:]
+        break
+
+    if relevant_lines is None:
+        raise ValueError(
+            f"description does not contain any relevant lines: {description}",
+        )
+
+    return REGEX_SENTENCE_DELIMITER.split("\n".join(relevant_lines), 1)[0] + "."
+
+
+def strip_techdocs_prefixes(description: str) -> str:
+    """
+    Removes all bold prefixes from the given description.
+
+    :param description: The description of a CLI argument.
+    :type description: str
+
+    :returns: The stripped description
+    :rtype: str
+    """
+    result_description = REGEX_TECHDOCS_PREFIX.sub(
+        "", description.lstrip()
+    ).lstrip()
+
+    return result_description
+
+
+def process_arg_description(description: str) -> Tuple[str, str]:
+    """
+    Processes the given raw request argument description into one suitable
+    for help pages, etc.
+
+    :param description: The original description for a request argument.
+    :type description: str
+
+    :returns: The description in Rich markup and original Markdown format.
+    :rtype: Tuple[str, str]
+    """
+
+    if description == "":
+        return "", ""
+
+    result = get_short_description(description)
+    result = strip_techdocs_prefixes(result)
+    result = result.replace("\n", " ").replace("\r", " ")
+
+    description, links = extract_markdown_links(result)
+
+    if len(links) > 0:
+        description += f" See: {'; '.join(links)}"
```

Comment:
Links are now extracted and appended to the end of an argument's description:

```
--region (required): The region where the Linode will be located. See: https://techdocs.akamai.com/linode-api/reference/get-regions
```

Previously links were embedded inline after the display text, but this made thing super unreadable with TechDocs due to long the new URLs are.